Sming/Services/libemqtt/libemqtt.h

```diff
@@ -168,6 +168,8 @@ typedef struct {
 	// Will topic
 	uint8_t will_retain;
 	uint8_t will_qos;
+	char* will_topic;
+	char* will_message;
```

Comment:
These two variables are defined as char* but you treat them like the other strings defined as fixed size char[] (i.e. you copy to them and memset them) but I don't see you ever malloc any memory to these pointers. Am I missing something?